hangman/lib/game/game.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Hangman.Game do
+  def new_game() do 
+    # function to filter alphabets
+    alphabets_only = fn x -> if Regex.match?(~r{[a-zA-Z]}, x),
+                               do: true,
+                               else: false
+                            end
+
+    # Get a random word from dictionary and store it as a list of characters
+    word = Dictionary.random_word()
+            |> String.split("")
+            |> Enum.filter(alphabets_only)
+
+    # Initialize the progress as blanks
+    letters =  Enum.map(word, fn _letter  -> "_" end)
+
+    %State{
+      game_state: :initializing,
+      turns_left: 7,
+      used: [],
+      last_guess: "",
+      letters: letters,
+      word: word
+    }
+  end
+
+  def tally(game) do
+    game
+      |> Map.from_struct()
+      |> Map.drop([:word])
+  end
+
+  def make_move(game, guess) do
+    if Enum.member?(game.used, guess) do
+      game = %State{ game |
+        game_state: :already_guessed
+      }
+      { game, tally(game) }
+    else
+      make_move(game, guess, :new_guess)
+    end
+  end
+  def make_move(game, guess, _new_guess) do
```

Comment:
If I were grading this, the size and complexity of this function would lose you marks.

Try to keep your functions to just a couple of lines, and think of ways of avoiding all the `if` statements. It takes a little more thinking up front, but results in code that "just works".

The trick it to ask yourself "what one thing does this function do?" Write that, in terms of other functions. Then, for each of those, ask "what does this do."